src/services/directoryServices/MediaDirectoryService.js

```diff
@@ -35,36 +36,32 @@ class MediaDirectoryService {
     if (!isMediaPathValid({ path: directoryName }))
       throw new BadRequestError("Invalid media folder name")
     const mediaType = directoryName.split("/")[0]
-    const { private: isPrivate } = await this.gitHubService.getRepoInfo(
-      sessionData
-    )
-    const files = (
-      await this.listWithDefault(sessionData, { directoryName })
-    ).filter(
-      (file) =>
-        (file.type === "file" || file.type === "dir") &&
-        file.name !== PLACEHOLDER_FILE_NAME
-    )
 
-    const resp = await Promise.all(
-      files.map((curr) => {
-        if (curr.type === "dir") {
-          return {
-            name: curr.name,
-            type: "dir",
-          }
-        }
-        return getMediaFileInfo({
-          file: curr,
-          siteName,
+    if (isGGSWhitelistedRepo(siteName)) {
```

Comment:
Hmm, is it possible to keep the check on whether the repo is whitelisted within RepoService? Because I was thinking RepoService should be the only one that is aware of what to do for whitelisted and non-whitelisted repositories. These downstream services would not have to worry about the underlying storage mechanisms.